### PR TITLE
feat(graphql for surveys): add mutation to create surveys

### PR DESCRIPTION
**Pull request by donni106**
- added acceptance of attributes that were missing for smooth creation with nested data
- added create mutation that resolves params to nested attributes passing them to create method for survey polls
- added field to mutation types to make the mutation available
- updated graphql schema files

SVA2-3

Bildschirmfoto 2021-06-17 um 16 15 39

Bildschirmfoto 2021-06-17 um 16 16 18

### CODE REVIEW

**Review by digorath — COMMENTED** [review]
✔️

File: app/graphql/mutations/create_survey_poll.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      params = params.merge(
+        questions_attributes: [{
+          title: params[:question_title],
+          response_options_attributes: params[:response_options]
+        }]
+      )
+      # delete params that shouldn't be passed to creation, because they are unknown to the model
+      params.delete(:question_title)
+      params.delete(:response_options)
+
```
Comment (marcometz) [inline app/graphql/mutations/create_survey_poll.rb:34]: Vorschlag:


```
question_title = params.delete(:question_title)
response_options = params.delete(:response_options)

params = params.merge(
questions_attributes: [{
  title: question_title,
  response_options_attributes: response_options
}]
)

```

dann haben wir keinen doppelten Zugriff auf die Params

**Review by marcometz — APPROVED** [review]

File: app/graphql/mutations/create_survey_poll.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    argument :title, GraphQL::Types::JSON, required: false
+    argument :description, GraphQL::Types::JSON, required: false
+    argument :date,
+             Types::InputTypes::DateInput,
+             required: false,
+             as: :date_attributes,
+             prepare: ->(date, _ctx) { date.to_h }
+    argument :questionTitle, GraphQL::Types::JSON, required: true
+    argument :response_options, [GraphQL::Types::JSON], required: true
+
```
Comment (donni106) [inline app/graphql/mutations/create_survey_poll.rb:14]: ha, und hier dann `[Types::QueryTypes::SurveyResponseOptionsType]`

Comment (donni106) [reply app/graphql/mutations/create_survey_poll.rb:14]: nee, geht nicht:

Bildschirmfoto 2021-06-18 um 09 53 30

File: app/graphql/mutations/create_survey_poll.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+module Mutations
+  class CreateSurveyPoll < BaseMutation
+    argument :title, GraphQL::Types::JSON, required: false
+    argument :description, GraphQL::Types::JSON, required: false
+    argument :date,
+             Types::InputTypes::DateInput,
+             required: false,
+             as: :date_attributes,
+             prepare: ->(date, _ctx) { date.to_h }
+    argument :questionTitle, GraphQL::Types::JSON, required: true
```
[the remark this answers is not in the document]
Comment (donni106) [reply app/graphql/mutations/create_survey_poll.rb]: ✔️

File: app/graphql/mutations/create_survey_poll.rb (hunk 1 above)
Comment (donni106) [reply app/graphql/mutations/create_survey_poll.rb:35]: ✔️

**Pull request merged**